statsmodels/regression/_prediction.py

```diff
@@ -156,20 +156,39 @@ def get_prediction(self, exog=None, transform=True, weights=None,
            (weights.ndim != 1 or weights.shape[0] == exog.shape[1])):
             raise ValueError('weights has wrong shape')
 
+    else:
+        # this handles the case where we don't want to apply weights
+        weights = 1.
     ### end
 
     if pred_kwds is None:
         pred_kwds = {}
     predicted_mean = self.model.predict(self.params, exog, **pred_kwds)
 
-    covb = self.cov_params()
-    var_pred_mean = (exog * np.dot(covb, exog.T).T).sum(1)
+    # handle var_pred_mean for the PHReg case
+    if 'pred_type' in pred_kwds:
+        cov_params = pred_kwds['cov_params']
+        if pred_kwds['pred_type'] == 'lhr':
```

Comment:
@lbybee can we avoid referring to `lhr` (duration-specific term) here in the base prediction code?  I think it would be better to have a subclass for phreg, and if `pred_type` is `lhr` we just invoke the superclass, otherwise we do phreg-specific things in the subclass.